llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2374,6 +2380,44 @@ class VPReductionPHIRecipe : public VPHeaderPHIRecipe,
   bool isInLoop() const { return IsInLoop; }
 };
 
+/// A recipe for forming partial reductions. In the loop, an accumulator and
+/// vector operand are added together and passed to the next iteration as the
+/// next accumulator. After the loop body, the accumulator is reduced to a
+/// scalar value.
+class VPPartialReductionRecipe : public VPRecipeWithIRFlags {
+  Instruction &ReductionInst;
```

Comment:
Looks like ReductionInst may only be used for the opcode in the latest version?. If that's the case, can just the opcode be stored?

Reply:
Can `getUnderlyingInst()` be used for cloning instead? There's little benefit to storing the instruction twice

Reply:
Great idea, thanks. Done.